mkosi/util.py

```diff
@@ -228,17 +228,18 @@ def flatten(lists: Iterable[Iterable[T]]) -> list[T]:
 
 class InvokingUser:
     @staticmethod
-    def _uid_from_env() -> Optional[str]:
-        return os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID")
+    def _uid_from_env() -> Optional[int]:
+        uid = os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID")
+        return int(uid) if uid is not None else None
 
     @classmethod
     def uid(cls) -> int:
-        return int(cls._uid_from_env() or os.getuid())
+        return cls._uid_from_env() or os.getuid()
 
     @classmethod
     def uid_gid(cls) -> tuple[int, int]:
-        if uid := cls._uid_from_env:
-            gid = int(os.getenv("SUDO_GID") or pwd.getpwuid(uid).pw_gid)
+        if (uid := cls._uid_from_env()) is not None:
+            gid = int(os.getenv("SUDO_GID", pwd.getpwuid(uid).pw_gid))
```

Comment:
Using `int(os.getenv("SUDO_GID") or pwd.getpwuid(uid).pw_gid)` meant that `pwd.getpwuid()`is not called when `$SUDO_GID` is set.